src/Common/src/System/Dynamic/Utils/ExpressionUtils.cs

```diff
@@ -171,7 +171,7 @@ public static Expression ValidateOneArgument(MethodBase method, ExpressionType n
             return arguments;
         }
 
-        public static void RequiresCanRead(Expression expression, string paramName, int idx)
+        public static void RequiresCanRead(Expression expression, string paramName, int idx = -1)
```

Comment:
Most of the callers seem to not pass `idx` and this helper is indeed called very often.
I think we should have an overload without `idx`.